src/Razor/src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -1651,7 +1651,18 @@ private void ParseAwaitExpression(SyntaxListBuilder<RazorSyntaxNode> builder, CS
             // Therefore, if we're top level then we want to act like an implicit expression,
             // otherwise just act as whatever we're contained in.
             var topLevel = transition != null;
-            if (topLevel)
+            if (!topLevel)
+            {
+                return;
+            }
+
+            if (At(CSharpKeyword.Foreach))
+            {
+                // C# 8 async streams. @await foreach (var value in asyncEnumerable) { .... }
+
+                ParseConditionalBlock(builder, transition);
```

Comment:
This means we will now support any keyword block that comes after `@await`. Is that intended? Or should we restrict this behavior to just `foreach`? I'm okay either way but we should make a clear decision.

Reply:
This wont hit for anything other than `foreach`. It's protected by the `if (At(CSharpKeyword.Foreach))` above.

Reply:
Oh duh. Missed it 😄